report/verification/mms/simple_transient.md

```diff
@@ -0,0 +1,329 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.7
+kernelspec:
+  display_name: vv-festim-report-env-festim-2
+  language: python
+  name: python3
+---
+
+# Simple transient diffusion case
+
+```{tags} 2D, MMS, transient
+```
+
+This is a simple transient MMS example.
+We will only consider diffusion of hydrogen in a unit square domain $\Omega$ at steady state with an homogeneous diffusion coefficient $D$.
+Moreover, a Dirichlet boundary condition will be assumed on the boundaries $\partial \Omega $.
+
+The problem is therefore:
+
+$$
+\begin{align}
+    &\nabla \cdot (D \ \nabla{c}) - \frac{\partial c}{\partial t} = -S  \quad \text{on }  \Omega  ; \ t\geq 0 \\
+    & c = c_0 \quad \text{on }  \partial \Omega ; \ t\geq 0 \\
+    & c = c_\mathrm{initial} \quad \text{on } \partial \Omega ; \ \text{at } t=0
+\end{align}
+$$(problem_simple_transient)
+
+The exact solution for mobile concentration is:
+
+$$
+\begin{equation}
+    c_\mathrm{exact} = 1 + 2 x^2 + 3 y^2 t + 2t
+\end{equation}
+$$(c_exact_simple_transient)
+
+```{note}
+We use a manufactured solution that varies linearly with time ($t^1$), as the backward Euler scheme provides an exact solution in this case.
+```
+
+Injecting {eq}`c_exact_simple_transient` in {eq}`problem_simple_transient`, we obtain the expressions of $S$, $c_0$, and $c_\mathrm{initial}$:
+
+\begin{align}
+    & S = 2 + 3 y^2 - (4 + 6t) D \\
+    & c_0 = c_\mathrm{exact} \\
+    & c_\mathrm{initial} = c_\mathrm{exact}(t=0)
+\end{align}
+
+We can then run a FESTIM model with these values and compare the numerical solution with $c_\mathrm{exact}$.
+
++++
+
+## FESTIM code
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+
+from mpi4py import MPI
+import festim as F
+from dolfinx.mesh import create_unit_square
+import numpy as np
+
+fenics_mesh = create_unit_square(MPI.COMM_WORLD, 100, 100)
+
+boundary = F.SurfaceSubdomain(id=1)
+H = F.Species("mobile", mobile=True)
+D = 2
+my_mat = F.Material(D_0=D, E_D=0)
+volume = F.VolumeSubdomain(id=1, material=my_mat)
+boundary = F.SurfaceSubdomain(id=1)
+
+exact_solution = (
+    lambda x, t: 1 + 2 * x[0] ** 2 + 3 * t * x[1] ** 2 + 2 * t
+)  # exact solution
+
+f = lambda x, t: 2 + 3 * x[1] ** 2 - (4 + 6 * t) * D
+
+final_time = 17
+slices = 4
+slice_size = final_time / slices
+my_milestones = list(np.linspace(slice_size, final_time, slices))
```

Comment:
```suggestion
my_milestones = np.linspace(slice_size, final_time, slices).tolist()
```

It will produce the same result, but this is more standard imo